### ISSUE TITLE
Flow monitoring proof of concept

### ISSUE THREAD

**Issue by mcangeli** [RaspberryPints/RaspberryPints#231]
Working off what @kerber had come up with, I managed to get the flow_monitor.py script talking to the flow monitors and to the database.

It updates the pours table with the amount poured in ounces. Once thats done, the vwGetTapAmountPoured view pulls it and converts it to gallons (I was having an issue working with decimals in python).

The index page is set to refresh every 10 minutes (though if you have a keyboard hooked up you can hit f5 when you're done pouring and see the change right away...)

this "fork" also has the untappd code in it and its been optimized for my monitor which runs 1280x720.

You can grab it here:
https://github.com/mcangeli/RaspberryPints/tree/Untappd

There are some things that need to be done prior to using the flow monitors:

You can setup the flow monitors with the arduino by following @kerber's post:
https://github.com/RaspberryPints/RaspberryPints/issues/194

You also need to install the mysqldb python module:

```
$ sudo apt-get install python-mysqldb
```

Once thats done, you need to edit the flow_monitor.py module and change the settings for the taps that map the taps to the pins. This is currently hard coded in the py file, my goal is to make this a config value in the database.

Also hard coded is the number of pulses per ounce. That is currently set at 165 (its an average and seems to be spot on). I plan to make this editable in the db as well.

There are some other settings (such as the info for untappd) that needs to be set thru the admin panel, right now, it defaults to my home brewery (yay me!).

If you have any questions or need help setting it up let me know. 

_Also, its important to know that I've made several changes to the DB for this so its not a simple upload and it works. This is best with a clean install at the moment as I have not created the change file for SQL_

**Comment by JayUnt** [comment]
I think we should have a convo on how all of this is going.  I know some of these things that you are working on with your code, @kerber and I were discussing as well.  I just dont want us all to work on the same thing different ways, and then just have a mix of different code that does the same thing.  

I know i am being a bit of a road block, with not having the re-write done yet, and im sorry about that.  I think, once i get to a certain point, where the main features are back in, we can work on the pours stuff, while the rest of the site is brought up to date (like the personalization/forgot password stuff).

Also, the schema script you are working on is based off of the old schema.  With the re-write there is a bunch of change to the db, so you script wont be in sync.

**Comment by mcangeli** [comment]
@JayUnt, I know that there've been changes to the DB which is why I am not pushing to commit my changes to the main git repo at this point. This was just me playing around with the current code base to see if I could make it work and for what I needed, it is working so there's a valid proof of concept version.

(this way people can see, yep, it really is possible for it to do what they say it can)

It was something for me to do while you convert the site to laravel as I'll have to rework all the stuff I've done for my screen and the untappd integration anyways. :)

**Comment by JayUnt** [comment]
Sounds good!

If you want to start adding your Untappd stuff to my branch, I think we are at a point where you can start.  I would suggest making an Untappd Model and Controller.  I have only made models that pull from the db and not a third party api, so you can try to figure that part out.

here are some links i quickly found:
http://stackoverflow.com/questions/19124018/laravel-retrieving-data-from-rest-api
http://forumsarchive.laravel.io/viewtopic.php?id=1466

**Comment by mcangeli** [comment]
thanks.

**Comment by JayUnt** [comment]
This might be useful too

https://github.com/PintLabs/Pintlabs_Service_Untappd

**Comment by mcangeli** [comment]
Thats the code I'm already using for the untappd api. ;)

**Comment by mcangeli** [comment]
Gah. You changed from gallons to liters.

**Comment by JayUnt** [comment]
Sweet!  We can throw that into the vendors folder and then have the Untappd model use it.  I'm thinking we would not have the Untappd model extend Eloquent, like the other are, and we can just put all of your getter functions in there as static functions. It then work similar to the other models.

Ahhh yeah, the liters change was on request from the kerbs.  I didn't have a strong opinion either way.  

At my company, we used to have a service that worked off of facebook's api.  And this was what it was like.  Every couple of months they would change some functionality just enough to mess up everything you worked on.  Needless to say, we dont use their api anymore.

**Comment by mcangeli** [comment]
ok... making a dev. subdomain on my pi so I can have my main one then my test one...

**Comment by mcangeli** [comment]
Also, I'm on google as mcangeli

it may be easier talking via hangouts.

**Comment by kerber** [comment]
The change to Metric is being done so that all the calculations are nice and clean. This has the added benefit of metric support, as all conversions can be done at the UI or API input layer based on the user preferences.

**Comment by Skemp45** [comment]
Old thread see no current value, if there is valuable info please add as a new issue with a descriptive title
